metisp/pymetis/src/pymetis/recipes/hci/metis_ifu_adi_cgrph.py

```diff
@@ -0,0 +1,166 @@
+"""
+Thi file is part of the METIS Pipeline.
+Copyright (C) 2024 European Southern Observatory
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+"""
+
+from pyesorex.parameter import ParameterList, ParameterEnum
+
+# import the dataitems we use
+from pymetis.classes.dataitems import DataItem, Hdu
+from pymetis.dataitems.distortion import IfuDistortionTable
+from pymetis.dataitems.ifu import IfuSciReducedCube
+#from pymetis.dataitems.hci import LmOffAxisPsfRaw, LmOnAxisPsfTemplate
+from pymetis.dataitems.hci.ifuHci import IfuRavcCalibrated, IfuCvcCalibrated, IfuAppCalibrated
+
+
+from pymetis.dataitems.hci.ifuHci import IfuRavcSciCentred, IfuRavcCentroidTab
+from pymetis.dataitems.hci.ifuHci import IfuRavcSciSpeckle, IfuRavcSciHifilt, IfuRavcSciDerotatedPsfsub
+from pymetis.dataitems.hci.ifuHci import IfuRavcSciDerotated
+from pymetis.dataitems.hci.ifuHci import IfuRavcSciContrastRadprof, IfuRavcSciContrastAdi, IfuRavcSciThroughput
+from pymetis.dataitems.hci.ifuHci import IfuRavcSciCoverage, IfuRavcSciSnr
+
+from pymetis.dataitems.hci.ifuHci import IfuCvcSciCentred, IfuCvcCentroidTab
+from pymetis.dataitems.hci.ifuHci import IfuCvcSciSpeckle, IfuCvcSciHifilt, IfuCvcSciDerotatedPsfsub
+from pymetis.dataitems.hci.ifuHci import IfuCvcSciDerotated
+from pymetis.dataitems.hci.ifuHci import IfuCvcSciContrastRadprof, IfuCvcSciContrastAdi, IfuCvcSciThroughput
+from pymetis.dataitems.hci.ifuHci import IfuCvcSciCoverage, IfuCvcSciSnr
+
+from pymetis.dataitems.hci.ifuHci import IfuAppSciCentred, IfuAppCentroidTab
+from pymetis.dataitems.hci.ifuHci import IfuAppSciSpeckle, IfuAppSciHifilt, IfuAppSciDerotatedPsfsub
+from pymetis.dataitems.hci.ifuHci import IfuAppSciDerotated
+from pymetis.dataitems.hci.ifuHci import IfuAppSciContrastRadprof, IfuAppSciContrastAdi, IfuAppSciThroughput
+from pymetis.dataitems.hci.ifuHci import IfuAppSciCoverage, IfuAppSciSnr
+from pymetis.dataitems.ifu.ifu import IfuScienceCubeCalibrated
+
+from pymetis.classes.recipes import MetisRecipe
+from pymetis.classes.prefab import RawImageProcessor
+from pymetis.classes.inputs import RawInput
+from pymetis.utils.dummy import create_dummy_header, create_dummy_image, create_dummy_table
+
+
+
+class MetisIfuRavcSciCalibrateImpl(RawImageProcessor):
```

Comment:
Given the name of this recipe, I suppose it also applies to the `CVC` coronograph. I don't think it is necessary to create a separate recipe for each coronograph, but not sure.